spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteManifestsAction.java

```diff
@@ -466,6 +474,310 @@ public void testRewriteLargeManifestsPartitionedTable() throws IOException {
     assertThat(newManifests).hasSizeGreaterThanOrEqualTo(2);
   }
 
+  @TestTemplate
+  public void testRewriteManifestsPartitionedTableWithInvalidClusteringColumns()
+      throws IOException {
+    PartitionSpec spec = PartitionSpec.builderFor(SCHEMA).identity("c1").bucket("c3", 10).build();
+    Map<String, String> options = Maps.newHashMap();
+    options.put(TableProperties.FORMAT_VERSION, String.valueOf(formatVersion));
+    options.put(TableProperties.SNAPSHOT_ID_INHERITANCE_ENABLED, snapshotIdInheritanceEnabled);
+    Table table = TABLES.create(SCHEMA, spec, options, tableLocation);
+
+    SparkActions actions = SparkActions.get();
+
+    //  c2 is not a partition column, cannot use for clustering
+    List<String> badClusteringKeys1 = ImmutableList.of("c1", "c2");
+    assertThatThrownBy(
+            () ->
+                actions
+                    .rewriteManifests(table)
+                    .rewriteIf(manifest -> true)
+                    .clusterBy(badClusteringKeys1)
+                    .execute())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage(
+            "Cannot use custom clustering to rewrite manifests '[c1, c2]'. All partition columns "
+                + "must be defined in the current partition spec: 0. Choose from the available "
+                + "partitionable columns: [c3_bucket, c1]");
+
+    // c3_bucket is the correct internal partition name to use, c3 is the untransformed column name,
+    // clusterBy() expects the hidden partition column names
+    List<String> badClusteringKeys2 = ImmutableList.of("c1", "c3");
```

Comment:
wrongFieldName

Reply:
In general we shouldn't use variables with a numerical suffix if we can give them a more meaningful name